### PR TITLE
TCVP-2307 insert email fails

### PR DESCRIPTION
**Pull request by lfdgov**
# Description

This PR includes the following proposed change(s):

- TCVP-2307

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/TrafficCourts/Workflow.Service/Mappings/Mapper.cs [hunk 1]
```diff
@@ -14,8 +14,17 @@ public static EmailHistory ToEmailHistory(DisputantEmailSent src)
         target.ToEmailAddress = src.Message?.To is not null ? src.Message.To : "unknown";
         target.SuccessfullySent = EmailHistorySuccessfullySent.Y;
         target.Subject = src.Message?.Subject is not null ? src.Message.Subject : "unknown";
-        target.HtmlContent = src.Message?.HtmlContent is not null ? src.Message.HtmlContent : "";
```
Comment (lfdgov) [inline src/backend/TrafficCourts/Workflow.Service/Mappings/Mapper.cs:17]: ords endpoint does some validation including that only one of htmlcontent or plaintextcontent can be non-null.  staff api was ensuring that both values were non-null thus the failure on every call

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/EmailHistoryController.java [hunk 2]
```diff
@@ -60,7 +61,20 @@ public List<EmailHistory> getEmailHistoryByTicketNumber(
...
-		logger.debug("POST /emailHistory called to save emailHistory: {}", StructuredArguments.fields(emailHistory));
-		return new ResponseEntity<Long>(emailHistoryService.insertEmailHistory(emailHistory), HttpStatus.OK);
+		logger.debug("POST /emailHistory called with: {}", StructuredArguments.fields(emailHistory));
+		// return bad request for invalid arguments
+		if (emailHistory.getOccamDisputeId() == null || emailHistory.getOccamDisputeId() == 0)
+			throw new IllegalArgumentException("Occam dispute id is zero or missing.");
+		else if (emailHistory.getHtmlContent() != null && emailHistory.getPlainTextContent() != null )
+			throw new IllegalArgumentException("Only one of plaintextcontent or htmlcontent can be supplied.");
+		else if (emailHistory.getSuccessfullySent() == null || emailHistory.getSuccessfullySent() == YesNo.UNKNOWN)
+			throw new IllegalArgumentException("parameter successfullySent has unknown value");
```
Comment (lfdgov) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/EmailHistoryController.java:71]: check for all illegal arguments that ords endpoint checks for so we dont get a 500 exception

File: src/backend/TrafficCourts/Workflow.Service/Mappings/Mapper.cs [hunk 3]
```diff
@@ -14,8 +14,17 @@ public static EmailHistory ToEmailHistory(DisputantEmailSent src)
...
-        target.PlainTextContent = src.Message?.TextContent is not null ? src.Message.TextContent : "";
+        // ensure only one of html content or plain text content is set
+        if (src.Message?.HtmlContent is not null && src.Message?.TextContent is not null)
```
Comment (pbolduc) [inline src/backend/TrafficCourts/Workflow.Service/Mappings/Mapper.cs:18]: I do not like having this logic at this layer. Emails can have both HTML and Plain Text. If the ORDS endpoint requires one or the other, we can move this logic further down the pipeline, like into the Oracle Data API. I believe at this point we are only sending plain text emails. We had a conversation with the product owner and using HTML emails add additional compilations for formatting.

**Review by KDNeufeld — APPROVED** [review]
Approved. But if we remove all the HTML representations of the emails, what will happen when the client wants to see structured emails?

**Pull request merged**